mongoose.c

```diff
@@ -1971,6 +1971,8 @@ struct threadparam {
 };
 
 static int wait_until_ready(sock_t sock, int for_read) {
+  if ( (sock < 0) || (sock >= FD_SETSIZE) )
```

Comment:
Windows - V547 Expression 'sock < 0' is always false. Unsigned type value is never < 0